lib/tasks_private/instance_types.rake

```diff
@@ -0,0 +1,81 @@
+namespace 'aws:extract' do
+  desc 'Get / renew instance_types'
+  task :instance_types => [:environment] do
+    # Instance Types provided by the AWS SDK do not have sufficient info for provisioning
+    #
+    # This task creates a db/fixtures/aws_instance_types.yml file with data from
+    #  curl https://raw.githubusercontent.com/powdahound/ec2instances.info/master/www/instances.json > lib/tasks/instance_types_data/instances.json
+    #  https://aws.amazon.com/ec2/instance-types/#instance-type-matrix > lib/tasks/instance_types_data/instance_types.csv
+    #  https://aws.amazon.com/ec2/previous-generation/#Previous_Generation_Instance_Details_and_Pricing_ > lib/tasks/instance_types_data/instance_types_previous.csv
+    #     converted to csv via http://www.convertcsv.com/html-table-to-csv.htm
+    #
+    # Other useful resources
+    #   http://aws.amazon.com/ec2/instance-types
+    #   http://docs.amazonwebservices.com/AWSEC2/latest/UserGuide/instance-types.html
+    #   http://aws.amazon.com/ec2/previous-generation
+    #   http://www.ec2instances.info/
+    #   http://docs.aws.amazon.com/AWSEC2/latest/UserGuide/t2-instances.html
+    #   http://docs.aws.amazon.com/AWSEC2/latest/UserGuide/c4-instances.html
+
+    def instances
+      require 'open-uri'
+      URI.open("https://raw.githubusercontent.com/powdahound/ec2instances.info/master/www/instances.json") do |io|
+        JSON.parse(io.read)
+      end
+    end
+
+    results = instances.map do |instance|
+      storage = instance["storage"] || {}
+      num_storage_devices = storage["devices"].to_i
+      storage_size        = storage["size"].to_f
+
+      storage_type = if storage["nvme_ssd"]
+                       "NVMe SSD"
+                     elsif storage["ssd"]
+                       "SSD"
+                     end
+
+      network_performance = if instance["network_performance"].match?(/(\d+ Gigabit)/)
+                              :very_high
+                            else
+                              instance["network_performance"].downcase.tr(' ', '_').to_sym
+                            end
+
+      result = {
+        :architecture            => instance["arch"].map(&:to_sym).sort,
+        :cluster_networking      => nil,
+        :current_generation      => instance["generation"] != "previous",
+        :current_version         => true,
+        :description             => instance["pretty_name"],
+        # TODO :ebs_only => ???
```

Comment:
Oh I see you already have it TODO'd

Reply:
Okay instance["storage"].nil? co-implies ebs-only :tada:

Reply:
I was worried I was going to have to pull down the https://aws.amazon.com/ec2/instance-types/ list to get this property but luckily not